### ISSUE TITLE
test(gemma4): synthetic golden checksum is not portable across AVX2 hosts (CI always red)

### ISSUE THREAD

**Issue by jamesburton** [jamesburton/dotLLM#107]
﻿## Problem
`SyntheticGemma4RegressionTests.Gemma4_Tiny_AllFeatures_And_DeterministicGolden` fails on CI (and has since it was introduced in #40). It is the only red test in the otherwise-green Integration Tests job, blocking a clean CI on every dev PR.

```
Assert.Equal() Failure: Values differ
Expected: 3290309287438143715   (0x2DA986DD36EB50E3)
Actual:   11562233381654802934  (0xA0754E0053B2B1F6)
```

## Root cause
The test hard-codes `Gemma4LastRowChecksum` = an FNV-1a over the **raw float32 bits** of the last logit row, captured on a developer AVX2 host, and asserts it whenever `Avx2.IsSupported`. But the CI runner is a *different* AVX2 microarchitecture/runtime: FMA contraction and BCL `MathF.Tanh`/`Exp` (final-logit softcap, softmax) are **not bit-identical across AVX2 hosts**, so the bit-exact golden is not portable even when gated on AVX2.

Evidence it is a stable host difference, not a regression:
- CI emits the **same** `checksum=0xA0754E0053B2B1F6` across runs on different commits/days (#101 Jun-25, #102 & #106 Jun-26).
- `argmax=144` (the documented "firm cross-environment golden") always matches.
- Run-to-run determinism (`c1==c2`) always passes.

## Fix
Remove the non-portable bit-exact checksum assertion. Keep the portable contracts the test already documents as the real gates: the argmax golden and run-to-run determinism. Retain the checksum value as a printed diagnostic reference.

## Acceptance
- Integration Tests job green on dev.
- Test still asserts: all Gemma-4 config features, argmax golden (144), run-to-run determinism, argmax in range.

**Comment by jamesburton** [comment]
Fixed by #108 (merged to dev). Integration Tests CI job now green.
